invoice.py

```diff
@@ -0,0 +1,29 @@
+# This file is part account_invoice_reference_required module for Tryton.
+# The COPYRIGHT file at the top level of this repository contains
+# the full copyright notices and license terms.
+from trytond.pool import PoolMeta, Pool
+from trytond.i18n import gettext
+from trytond.exceptions import UserWarning
+
+
+class Invoice(metaclass=PoolMeta):
+    __name__ = 'account.invoice'
+
+    @classmethod
+    def set_number(cls, invoices):
+        pool = Pool()
+        Warning = pool.get('res.user.warning')
+
+        super().set_number(invoices)
+
+        for invoice in invoices:
+            if invoice.type != 'in':
+                continue
+            if not invoice.reference or invoice.reference == '':
```

Comment:
The `invoice.reference == ''`will never evaluate to True if `not invoice.reference` evaluates to `False`so it is useless.

It should simply be: `if not invoice.reference`.